dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp

```diff
@@ -95,23 +95,54 @@ void KVStore::checkAndApplyPreHandledSnapshot(const RegionPtrWrap & new_region,
             if (overlapped_region.first != region_id)
             {
                 auto state = getProxyHelper()->getRegionLocalState(overlapped_region.first);
-                if (state.state() != raft_serverpb::PeerState::Tombstone)
+                auto extra_msg = fmt::format(
+                    "state={}, tiflash_state={}, new_region_state={}",
+                    state.ShortDebugString(),
+                    overlapped_region.second->getMeta().getRegionState().getBase().ShortDebugString(),
+                    new_region->getMeta().getRegionState().getBase().ShortDebugString());
+                if (state.state() == raft_serverpb::PeerState::Tombstone)
                 {
-                    throw Exception(
-                        ErrorCodes::LOGICAL_ERROR,
-                        "range of region_id={} is overlapped with region_id={}, state: {}",
+                    LOG_INFO(
+                        log,
+                        "range of region_id={} is overlapped with `Tombstone` region_id={}, {}",
                         region_id,
                         overlapped_region.first,
-                        state.ShortDebugString());
+                        extra_msg);
+                    handleDestroy(overlapped_region.first, tmt, task_lock);
+                }
+                else if (state.state() == raft_serverpb::PeerState::Applying)
+                {
+                    auto r = RegionRangeKeys::makeComparableKeys(
+                        TiKVKey::copyFrom(state.region().start_key()),
+                        TiKVKey::copyFrom(state.region().end_key()));
+
+                    if (RegionsRangeIndex::isRangeOverlapped(new_range->comparableKeys(), r))
+                    {
+                        throw Exception(
```

Comment:
Add comments about
* why we need to compare the keyrange of `new_region` and the keyrange of `state`
* in what situation the second comparison can pass or throw an exception

Reply:
Fixed